src/core/public/application/scoped_history.ts

```diff
@@ -219,12 +219,19 @@ export class ScopedHistory<HistoryLocationState = unknown>
 
   /**
    * Creates an href (string) to the location.
+   * If `prependBasePath` is true (default), it will prepend the location's path with the scoped history basePath.
    *
    * @param location
+   * @param prependBasePath
    */
-  public createHref = (location: LocationDescriptorObject<HistoryLocationState>): Href => {
+  public createHref = (
+    location: LocationDescriptorObject<HistoryLocationState>,
+    prependBasePath: boolean = true
+  ): Href => {
     this.verifyActive();
-    return this.parentHistory.createHref(location);
+    return this.parentHistory.createHref(
+      prependBasePath ? this.prependBasePath(location) : location
+    );
```

Comment:
I think that in some case, plugins might want to be able to use `createHref` to create href pointing outside of their basePath, so I added this optional parameter.

Being optional, it preserves the `History` interface